app/src/main/java/com/mainApp/yelp_mini/model/repo/APIClass.kt

```diff
@@ -0,0 +1,140 @@
+package com.mainApp.yelp_mini.model.repo
+
+import android.annotation.SuppressLint
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import com.mainApp.yelp_mini.model.data.YelpRestaurantDetail
+import com.mainApp.yelp_mini.model.data.YelpRestaurantReviews
+import com.mainApp.yelp_mini.model.data.YelpSearchResult
+import com.mainApp.yelp_mini.model.retroServices.RetroInstance
+import com.mainApp.yelp_mini.model.retroServices.YelpService
+import com.mainApp.yelp_mini.model.util.API_KEY
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+private const val TAG = "APIClass"
+
+class APIClass {
+
+    private val restaurantsDetailList: MutableLiveData<YelpRestaurantDetail?> = MutableLiveData()
+    private val restaurantsReviewList: MutableLiveData<YelpRestaurantReviews?> = MutableLiveData()
+    private val restaurants: MutableLiveData<YelpSearchResult?> = MutableLiveData()
+
+    fun restaurantReviewAPICall(restaurantID: String): MutableLiveData<YelpRestaurantReviews?> {
```

Comment:
Don't return a LiveData from APIClass. Instead, you can return the result in a list or an object.